crates/bevy_core/src/time/time.rs

```diff
@@ -143,8 +155,16 @@ impl Time {
     }
 }
 
-pub(crate) fn time_system(mut time: ResMut<Time>) {
-    time.update();
+/// The system used to update the time. If there is a render world the time is sent from
+/// there to this system through channels. Otherwise the time is updated in this system.
+pub(crate) fn time_system(mut time: ResMut<Time>, time_recv: Option<Res<TimeReceiver>>) {
+    if let Some(time_recv) = time_recv {
+        if let Ok(new_time) = time_recv.0.try_recv() {
```

Comment:
Worth thinking about any possible race conditions, when actual parallel pipelined rendering is enabled. Is there ever a scenario where this system might run when there is no value in the channel yet?

Given that this system runs at the start of the app schedule, and that the value is sent from the Render stage in the render schedule, it seems unlikely, but I don't think there is anything that *guarantees* it.

It seems to me that the non-blocking `try_recv` call, and just simply not doing anything if it fails (skipping the Time update), might not be correct behavior.

At the very least, we should panic with a message if it fails.

Reply:
>  Is there ever a scenario where this system might run when there is no value in the channel yet?
- it does happen every time the app is run on the first frame. This will turn into the first 2 frames in pipelined rendering.
- Once we have pipelining we might end up with 2 messages on the channel if rendering finishes before the time system is run. I should increase the bounds to 2 for that scenario.
- I don't think the opposite problem is possible normally. That the channel can be empty when you check it. This is because we have to sync on extract and the next app schedule can't be run until the previous render is done.

> It seems to me that the non-blocking try_recv call, and just simply not doing anything if it fails (skipping the Time update), might not be correct behavior.
>
> At the very least, we should panic with a message if it fails.

I went back and forth on adding a log message. I decided not to in case there are situations where people are running the time system multiple times per a render as part of another schedule. I do agree that if there's an unexpected skip that is potentially "very bad". So I'm not against adding a log message if others think it's a good idea.

Reply:
I'm on board with adding a log message. Users who are doing something strange can filter it themselves.

Reply:
When we have pipelining, the execution of this will race with the renderer system writing the value into it. When adding pipelining, as in, I would not say it has to be solved in this particular PR, we would have to consider how to manage that situation without making this do a blocking wait.

Reply:
We should be able to do that by just telling the time update system to not check the channel for the first 2 frames. I'll add a todo for that, since we need a way of detecting if the rendering is pipelined or not.